### PR TITLE
Dr 1405 Allow downgrade

### PR DESCRIPTION
**Pull request by mvillaseco**
# Overview
 Allow the users to do a Downgrade inside the webapp.

# Changes
 - Change change plan buttons.
 - Go to Confirmation page without any billing data.
 - Add Popup to confirm the downgrade.

### CODE REVIEW

File: src/wwwroot/partials/settings/billing.html [hunk 1]
```diff
@@ -200,8 +200,8 @@ <h3 class="special">{{ 'payment_method_title' | translate }}</h3>
       <div class="detached--error-container" ng-class="vm.submitted && vm.paymentFailure ? 'show':'hide'">
         <span class="detached--error">{{'validation_error_payment_failure' | translate}}</span>
       </div>
-      <button class="button button--small button--white modify-button" ng-click="vm.showConfirmation = false;vm.paymentFailure = false;">{{'billing_change_info_button' | translate}}</button>
-      <button class="button button--small buy-button" ng-click="vm.submitBillingPayment()">{{'buy_text' | translate}}</button>
+      <button class="button button--small button--white modify-button" ng-click="vm.cancelAction()"><span ng-if="!vm.downgrade">{{'billing_change_info_button' | translate}}</span><span ng-if="vm.downgrade">Volver a My Plan</span></button>
+      <button class="button button--small buy-button" ng-click="vm.submitBillingPayment()"><span ng-if="!vm.downgrade">{{'buy_text' | translate}}</span><span ng-if="vm.downgrade">Confirmar Downgrade</span></button>
```
Comment (alegarciamina) [inline src/wwwroot/partials/settings/billing.html]: We should add translations for "Volver a My Plan" and "Confirmar Downgrade" texts.

Comment (mvillaseco) [reply src/wwwroot/partials/settings/billing.html]: Good Catch!

File: src/wwwroot/index.html [hunk 2]
```diff
@@ -87,13 +87,14 @@
...
   <script src="controllers/modals/GeneralTemplateCtrl.js"></script>
+  <script src="controllers/modals/ConfirmInputTemplate.js"></script>
   <script src="controllers/SettingsCtrl.js"></script>
   <script src="controllers/DomainManagerCtrl.js"></script>
   <script src="controllers/dKimConfigCtrl.js"></script>
   <script src="controllers/RegistrationCtrl.js"></script>
   <script src="controllers/MyProfileCtrl.js"></script>
   <script src="controllers/PlanCtrl.js"></script>
-  <script src="controllers/BillingCtrl.js"></script>
+  <script src="controllers/BillingCtrl.js"></script>
```
Comment (andresmoschini) [inline src/wwwroot/index.html]: Very, very minor, could you remove this change?

File: src/wwwroot/locales/en-translation.js [hunk 3]
```diff
@@ -117,6 +117,7 @@
   "validation_error_mask": "Invalid Format.",
   "validation_error_invalid_card_number": "Invalid Credit Card number.",
   "validation_error_payment_failure": "We couldn't process the payment, please check your information.",
+  "validation_error_pattern":"Please check the text",
```
Comment (andresmoschini) [inline src/wwwroot/locales/en-translation.js]: Very, very minor, could you add the missed space after `:`?

File: src/wwwroot/locales/es-translation.js [hunk 4]
```diff
@@ -129,6 +129,7 @@
   "validation_error_mask": "Escribe un formato válido.",
   "validation_error_invalid_card_number": "Escribe una tarjeta de crédito válida.",
   "validation_error_payment_failure": "No pudimos realizar el pago, por favor revisa tu información.",
+  "validation_error_pattern":"Por favor revisa el texto ingresado",
```
Comment (andresmoschini) [inline src/wwwroot/locales/es-translation.js]: Very, very minor, could you add the missed space after `:`?

File: src/wwwroot/services/settings.js [hunk 5]
```diff
@@ -164,6 +164,22 @@
...
+
+    return $http({
+      actionDescription: 'action_billing_downgrade',
+      tryHandleError: function(rejection){ return tryHandleErrorBilling(rejection, onExpectedError); },
+      method: 'PUT',
+      data: agreement,
+      url: url
+    });
+    }
+
```
Comment (andresmoschini) [inline src/wwwroot/services/settings.js]: Very minor, wrong indentation? or it is me?

```js
    function downgrade(agreement, onExpectedError) {
        var url = RELAY_CONFIG.baseUrl
            + '/accounts/'
            + auth.getAccountName()
            + '/agreements'
            +  '/next';
        ^^^^
    return $http({
        actionDescription: 'action_billing_downgrade',
        tryHandleError: function(rejection){ return tryHandleErrorBilling(rejection, onExpectedError); },
        method: 'PUT',
        data: agreement,
        url: url
    });
}
^

```

**Comment by andresmoschini** [comment]
Nice changes @mvillaseco, It looks simpler that I expected, great!

**Pull request merged**